packages/BetterPhpDocParser/PhpDoc/ArrayItemNode.php

```diff
@@ -13,31 +13,21 @@ final class ArrayItemNode implements PhpDocTagValueNode, Stringable
 {
     use NodeAttributes;
 
-    /**
-     * @param String_::KIND_*|null $kindValueQuoted
-     */
     public function __construct(
         public mixed $value,
-        public mixed $key,
-        public int|null $kindValueQuoted = null,
-        public int|null $kindKeyQuoted = null,
+        public mixed $key = null,
     ) {
```

Comment:
I prefer to add fallback on string value instead of `StringNode` object so you don't need to use fork, and update rector-doctrine and rector-symfony later, the consumer will not BC break as well:

```suggestion
    ) {
    if (is_string($value)) {
        $this->value = new \Rector\BetterPhpDocParser\PhpDoc\StringNode($value);
    }
```

Reply:
@samsonasik I don't think this will work.

`StringNode` represent a string value in Doctrine Annotation, being quoted with double quotes.
If we go with your proposal, something like `@Foo(bar=baz)` will become parsed and printed as `@Foo(bar="baz")`.
Question would be is `@Foo(bar=baz)` valid?
I think it is at least for PHP constants, like in `@Foo(bar=SomeClass::MY_CONST)`.
`SomeClass::MY_CONST` is not same as `"SomeClass::MY_CONST"`

About the use of my fork, my guess is that this is temporary to show that all tests pass, end even if PRs are merged with it I guess there must be following PRs to clean that :)

Reply:
Side note: @TomasVotruba as rector-src currently `require` the packages, wouldn't a monorepo workflow be better for the Rector case? as packages are not optional for end users.
It would have simplified a lot things :)

Reply:
@samsonasik package rules will need to be updated anyway to handle properly the new node. not sure it's worth it.
but if you really want I can give it a try, do you? :)

Reply:
I can see that `ArrayItemNode` is used in rector extensions so keep the functionality is a prefered imo as it is not internal.

Reply:
could you point me to the extensions please?

Reply:
It was a private consultancy :), this change definitely break, using new value object is ok for me as far as there is fallback behaviour, or if it will land in 1.x.

Reply:
@samsonasik here it's a BC break yes, but it's even more a bug fix on Doctrine Annotation parsing.
We don't want a BC layer for something broken, right? :)
To me, still being on 0.x allows us to break things and polish everything before a 1.0 stable release. See all recent rules removals, maybe some users were using them, and we know they did.
The 0.x version is what communicates to end users that breaks may happen, even if we do our best to minimize them.

Reply:
Ok,  I will let @TomasVotruba decide then ;)